app/bundles/MessengerBundle/MessageHandler/PageHitNotificationHandler.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\MessengerBundle\MessageHandler;
+
+use Mautic\CoreBundle\Helper\DateTimeHelper;
+use Mautic\LeadBundle\Entity\LeadRepository;
+use Mautic\MessengerBundle\Exceptions\InvalidPayloadException;
+use Mautic\MessengerBundle\Exceptions\MauticMessengerException;
+use Mautic\MessengerBundle\Factory\MessengerRequestFactory;
+use Mautic\MessengerBundle\Message\PageHitNotification;
+use Mautic\PageBundle\Entity\Hit;
+use Mautic\PageBundle\Entity\HitRepository;
+use Mautic\PageBundle\Entity\PageRepository;
+use Mautic\PageBundle\Entity\RedirectRepository;
+use Mautic\PageBundle\Model\PageModel;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Messenger\Handler\Acknowledger;
+use Symfony\Component\Messenger\Handler\MessageSubscriberInterface;
+
+class PageHitNotificationHandler implements MessageSubscriberInterface
+{
+    public function __construct(private PageRepository $pageRepository, private HitRepository $hitRepository, private LeadRepository $leadRepository, private LoggerInterface $logger, private RedirectRepository $redirectRepository, private PageModel $pageModel)
+    {
+    }
+
+    /** @throws MauticMessengerException */
+    public function __invoke(PageHitNotification $message, Acknowledger $ack = null): void
+    {
+        try {
+            $parsed = $this->parseMessage($message);
+            $this->pageModel->processPageHit(...$parsed);
+        } catch (InvalidPayloadException $payloadException) {
+            $this->logger->error('Invalid payload #'.$message->getHitId().' '.$payloadException->getMessage());
+
+            throw $payloadException;
+        } catch (\Exception $exception) {
+            $this->logger->error($exception->getMessage(), (array) $exception);
+            throw new MauticMessengerException($exception->getMessage(), 400, $exception);
+        }
+        $this->logger->info('processed page hit #'.$message->getHitId());
```

Comment:
it could but it doesn't need to.

Reply:
There is a difference though. With this state, it will write `processed page hit #` to the logs no matter if it fails or not. I think it should write it only if it succeeds. Because if it fails then the page hit was not processed, right?

Reply:
either exception is thrown or info is written. never both.

Reply:
I will get rid of the MauticMessengerException though.

Reply:
You are right. I missed that.